inc/commonitilobject.class.php

```diff
@@ -7020,7 +7021,16 @@ function showTimeline($rand) {
                $entity = $this->getEntityID();
                if (Entity::getUsedConfig('anonymize_support_agents', $entity)
                   && Session::getCurrentInterface() == 'helpdesk'
-                  && ITILFollowup::getById($item_i['id'])->isFromSupportAgent()
+                  && (
+                     $item['type'] == "Solution"
+                     || is_subclass_of($item['type'], "CommonITILTask")
+                     || ($item['type'] == "ITILFollowup"
+                        && ITILFollowup::getById($item_i['id'])->isFromSupportAgent()
+                     )
+                     || ($item['type'] == "Document_Item"
+                        && Document_Item::getById($item_i['documents_item_id'])->isFromSupportAgent()
+                     )
+                  )
```

Comment:
We always anonymize Solution and tasks since self service users can't create them.

For followups and documents, check the author.